Upstream-Status: Inappropriate [cross compile specific]

--- php-8.4.1/scripts/php-config.in.old	2017-01-18 19:17:47.000000000 -0500
+++ php-8.4.1/scripts/php-config.in	2017-04-11 09:36:55.171279029 -0400
@@ -6,7 +6,7 @@
 exec_prefix="@exec_prefix@"
 version="@PHP_VERSION@"
 vernum="@PHP_VERSION_ID@"
-include_dir="@includedir@/php"
+include_dir="@includedir@/php@program_suffix@"
 lib_dir="@orig_libdir@"
 includes="-I$include_dir -I$include_dir/main -I$include_dir/TSRM -I$include_dir/Zend -I$include_dir/ext -I$include_dir/ext/date/lib"
 ldflags="@PHP_LDFLAGS@"
