Upstream-Status: Backport

--- mcrypt-1.0.7.old/mcrypt.c	2024-11-22 14:29:54.565751206 +0100
+++ mcrypt-1.0.7/mcrypt.c	2024-11-22 14:32:10.493262288 +0100
@@ -38,7 +38,11 @@
 #include "php_ini.h"
 #include "php_globals.h"
 #include "ext/standard/info.h"
+#if PHP_VERSION_ID >= 80400
+#include "ext/random/php_random.h"
+#else
 #include "ext/standard/php_rand.h"
+#endif
 #include "zend_smart_str.h"
 #include "php_mcrypt_filter.h"
 
@@ -1414,7 +1418,11 @@
 	} else {
 		n = (int)size;
 		while (size) {
+#if PHP_VERSION_ID >= 80400
+			iv[--size] = (char) (255.0 * php_mt_rand() / RAND_MAX);
+#else
 			iv[--size] = (char) (255.0 * php_rand() / RAND_MAX);
+#endif
 		}
 	}
 	RETVAL_STRINGL(iv, n);
