diff -durpN php-7.0.33.orig/acinclude.m4 php-7.0.33/acinclude.m4
--- php-7.0.33.orig/acinclude.m4	2024-05-14 10:27:12.846529582 +0200
+++ php-7.0.33/acinclude.m4	2024-05-14 10:30:14.357132972 +0200
@@ -1808,7 +1808,7 @@ main() {
 ], [
   cookie_io_functions_use_off64_t=no
 ], [
-  cookie_io_functions_use_off64_t=no
+  cookie_io_functions_use_off64_t=yes
 ])
     
     else
diff -durpN php-7.0.33.orig/main/streams/cast.c php-7.0.33/main/streams/cast.c
--- php-7.0.33.orig/main/streams/cast.c	2018-12-04 18:22:44.000000000 +0100
+++ php-7.0.33/main/streams/cast.c	2024-05-14 10:30:14.358132981 +0200
@@ -104,7 +104,7 @@ static ssize_t stream_cookie_writer(void
 }
 
 # ifdef COOKIE_SEEKER_USES_OFF64_T
-static int stream_cookie_seeker(void *cookie, __off64_t *position, int whence)
+static int stream_cookie_seeker(void *cookie, off64_t *position, int whence)
 {
 
 	*position = php_stream_seek((php_stream *)cookie, (zend_off_t)*position, whence);
