diff -durpN php-7.0.17.orig/ext/standard/config.m4 php-7.0.17/ext/standard/config.m4
--- php-7.0.17.orig/ext/standard/config.m4	2017-03-20 05:39:08.750168142 -0400
+++ php-7.0.17/ext/standard/config.m4	2017-03-20 05:41:01.839671978 -0400
@@ -349,9 +349,8 @@ dnl Check for available functions
 dnl
 dnl log2 could be used to improve the log function, however it requires C99. The check for log2 should be turned on,
 dnl as soon as we support C99.
-AC_CHECK_FUNCS(getcwd getwd asinh acosh atanh log1p hypot glob strfmon nice fpclass mempcpy strpncpy)
+AC_CHECK_FUNCS(getcwd getwd asinh acosh atanh log1p hypot glob strfmon nice fpclass mempcpy strpncpy fnmatch)
 AC_CHECK_DECLS([isnan, isinf], [], [], [[#include <math.h>]])
-AC_FUNC_FNMATCH	
 
 dnl
 dnl Check if there is a support means of creating a new process
