Upstream-Status: Inappropriate [cross compiler specific]

--- inotify-0.1.6/config.m4~	2012-05-04 05:51:20.000000000 -0400
+++ inotify-0.1.6/config.m4	2016-07-04 04:26:28.992903528 -0400
@@ -6,15 +6,7 @@
 
 if test "$PHP_INOTIFY" != "no"; then
 
-  AC_TRY_RUN([
-   #include <sys/inotify.h>
-   void testfunc(int (*passedfunc)()) {
-   }
-   int main() {
-    testfunc(inotify_init);
-    return 0;
-   }
-  ],[],[
+  AC_CHECK_FUNCS(inotify_init, [], [
    AC_MSG_ERROR(Your system does not support inotify)
   ])
 
