diff -durpN php-7.1.27.orig/acinclude.m4 php-7.1.27/acinclude.m4
--- php-7.1.27.orig/acinclude.m4	2019-03-26 08:13:34.903013732 -0400
+++ php-7.1.27/acinclude.m4	2019-03-26 08:14:07.728143081 -0400
@@ -1918,13 +1918,6 @@ AC_DEFUN([PHP_CHECK_FUNC_LIB],[
   ])
 
   if test "$found" = "yes"; then
-    ac_libs=$LIBS
-    LIBS="$LIBS -l$2"
-    AC_TRY_RUN([main() { return (0); }],[found=yes],[found=no],[found=no])
-    LIBS=$ac_libs
-  fi
-
-  if test "$found" = "yes"; then
     PHP_ADD_LIBRARY($2)
     PHP_DEF_HAVE($1)
     PHP_DEF_HAVE(lib$2)
