Upstream-Status: Inappropriate [cross compile specific]

diff -up ./sapi/embed/config.m4.embed ./sapi/embed/config.m4
--- ./sapi/embed/config.m4.embed	2025-03-26 07:14:03.223791152 +0100
+++ ./sapi/embed/config.m4	2025-03-26 07:16:43.185955159 +0100
@@ -15,7 +15,8 @@ if test "$PHP_EMBED" != "no"; then
         SAPI_SHARED="libs/libphp.dylib"
         PHP_EMBED_TYPE=shared-dylib
       ], [PHP_EMBED_TYPE=shared])
-      INSTALL_IT="\$(mkinstalldirs) \$(INSTALL_ROOT)\$(orig_libdir); \$(INSTALL) -m 0755 $SAPI_SHARED \$(INSTALL_ROOT)\$(orig_libdir)"
+      EXTRA_LDFLAGS="$EXTRA_LDFLAGS -release \$(PHP_MAJOR_VERSION).\$(PHP_MINOR_VERSION)"
+      INSTALL_IT="\$(mkinstalldirs) \$(INSTALL_ROOT)\$(libdir); \$(LIBTOOL) --mode=install \$(INSTALL) -m 0755 \$(OVERALL_TARGET) \$(INSTALL_ROOT)\$(libdir)"
     ],
     [static], [
       LIBPHP_CFLAGS="-static"
diff -up ./scripts/php-config.in.embed ./scripts/php-config.in
--- ./scripts/php-config.in.embed	2025-03-25 19:59:01.000000000 +0100
+++ ./scripts/php-config.in	2025-03-26 07:14:03.223870699 +0100
@@ -19,7 +19,7 @@ exe_extension="@EXEEXT@"
 php_cli_binary=NONE
 php_cgi_binary=NONE
 configure_options="@CONFIGURE_OPTIONS@"
-php_sapis="@PHP_INSTALLED_SAPIS@"
+php_sapis="apache2handler litespeed fpm phpdbg @PHP_INSTALLED_SAPIS@"
 ini_dir="@EXPANDED_PHP_CONFIG_FILE_SCAN_DIR@"
 ini_path="@EXPANDED_PHP_CONFIG_FILE_PATH@"
 php_embed_type="@PHP_EMBED_TYPE@"
