Upstream-Status: Inappropriate [cross compile specific]

diff -durpN php-7.3.0/pear/Makefile.frag php-7.3.0.new/pear/Makefile.frag
--- php-7.3.0/pear/Makefile.frag	2019-01-04 08:39:10.475516568 -0500
+++ php-7.3.0.new/pear/Makefile.frag	2019-01-04 09:12:25.670213378 -0500
@@ -12,7 +12,7 @@ PEAR_SUFFIX = -ds a$(program_suffix)
 PEAR_INSTALLER_URL = https://pear.php.net/install-pear-nozlib.phar
 
 install-pear-installer: $(SAPI_CLI_PATH)
-	@$(PHP_NATIVE_DIR)/php$(program_suffix) $(PEAR_INSTALL_FLAGS) $(builddir)/install-pear-nozlib.phar -d "$(peardir)" -b "$(bindir)" ${PEAR_PREFIX} ${PEAR_SUFFIX}
+	@$(PHP_NATIVE_DIR)/php$(program_suffix) $(PEAR_INSTALL_FLAGS) $(builddir)/install-pear-nozlib.phar -d "$(INSTALL_ROOT)$(peardir)" -b "$(INSTALL_ROOT)$(bindir)" ${PEAR_PREFIX} ${PEAR_SUFFIX}
 
 install-pear:
 	@echo "Installing PEAR environment:      $(INSTALL_ROOT)$(peardir)/"
