diff -durpN php-8.0.0.orig/ext/opcache/jit/Makefile.frag php-8.0.0/ext/opcache/jit/Makefile.frag
--- php-8.0.0.orig/ext/opcache/jit/Makefile.frag	2020-11-24 12:04:03.000000000 -0500
+++ php-8.0.0/ext/opcache/jit/Makefile.frag	2020-12-15 09:13:53.385857004 -0500
@@ -3,7 +3,7 @@ $(builddir)/minilua: $(srcdir)/jit/dynas
 	$(CC) $(srcdir)/jit/dynasm/minilua.c -lm -o $@
 
 $(builddir)/jit/zend_jit_x86.c: $(srcdir)/jit/zend_jit_x86.dasc $(srcdir)/jit/dynasm/*.lua $(builddir)/minilua
-	$(builddir)/minilua $(srcdir)/jit/dynasm/dynasm.lua  $(DASM_FLAGS) -o $@ $(srcdir)/jit/zend_jit_x86.dasc
+	php-native-minilua $(srcdir)/jit/dynasm/dynasm.lua  $(DASM_FLAGS) -o $@ $(srcdir)/jit/zend_jit_x86.dasc
 
 $(builddir)/jit/zend_jit.lo: \
 	$(builddir)/jit/zend_jit_x86.c \
