diff -durpN php-7.2.34.orig/acinclude.m4 php-7.2.34/acinclude.m4
--- php-7.2.34.orig/acinclude.m4	2024-05-14 11:43:45.643455212 +0200
+++ php-7.2.34/acinclude.m4	2024-05-14 11:44:58.179210403 +0200
@@ -658,9 +658,18 @@ AC_DEFUN([PHP_SET_LIBTOOL_VARIABLE],[
 ])
 
 dnl -------------------------------------------------------------------------
-dnl Wrapper macros for AC_ARG_WITH / AC_ARG_ENABLE
+dnl Wrapper macros for AC_ARG_WITH / AC_ARG_ENABLE / AC_TRY_RUN
 dnl -------------------------------------------------------------------------
 
+dnl PHP_TRY_RUN
+AC_DEFUN([PHP_TRY_RUN],[
+  if test "x$PHP_CROSSCOMPILING" = "xyes" ; then
+    AC_LINK_IFELSE(AC_LANG_SOURCE($1),$2,$3)
+  else
+    AC_TRY_RUN([$1],$2,$3,$4)
+  fi
+])
+
 dnl PHP_ARG_ANALYZE_EX
 dnl internal
 AC_DEFUN([PHP_ARG_ANALYZE_EX],[
@@ -1205,7 +1214,7 @@ dnl
 dnl PHP_DOES_PWRITE_WORK
 dnl internal
 AC_DEFUN([PHP_DOES_PWRITE_WORK],[
-  AC_TRY_RUN([
+  PHP_TRY_RUN([
 #include <stdlib.h>
 #include <sys/types.h>
 #include <sys/stat.h>
@@ -1236,7 +1245,7 @@ dnl PHP_DOES_PREAD_WORK
 dnl internal
 AC_DEFUN([PHP_DOES_PREAD_WORK],[
   echo test > conftest_in
-  AC_TRY_RUN([
+  PHP_TRY_RUN([
 #include <stdlib.h>
 #include <sys/types.h>
 #include <sys/stat.h>
diff -durpN php-7.2.34.orig/configure.ac php-7.2.34/configure.ac
--- php-7.2.34.orig/configure.ac	2024-05-14 11:43:45.643455212 +0200
+++ php-7.2.34/configure.ac	2024-05-14 11:44:29.800914948 +0200
@@ -209,6 +209,9 @@ PHP_ARG_ENABLE(rpath, whether to enable
 [  --disable-rpath         Disable passing additional runtime library
                           search paths], yes, no)
 
+PHP_ARG_ENABLE(crosscompiling, whether to enable cross compiling,
+[  --enable-crosscompiling Enable cross compiling], yes, no)
+
 dnl check for -R, etc. switch
 PHP_RUNPATH_SWITCH
 
diff -durpN php-7.2.34.orig/ext/curl/config.m4 php-7.2.34/ext/curl/config.m4
--- php-7.2.34.orig/ext/curl/config.m4	2024-05-14 11:43:45.644455222 +0200
+++ php-7.2.34/ext/curl/config.m4	2024-05-14 11:44:29.801914959 +0200
@@ -149,14 +149,18 @@ int main(int argc, char *argv[])
   return 1;
 }
 ], [
-      AC_MSG_RESULT([yes])
+      ext_curl_curl_gnutls=yes
+    ], [
+      ext_curl_curl_gnutls=no
+    ], [
+      dnl Default CURL build in Yocto 2.2
+      ext_curl_curl_gnutls=yes
+    ])
+    AC_MSG_RESULT([$ext_curl_curl_gnutls])
+    AS_IF([test x$ext_curl_curl_gnutls = xyes],[
       AC_CHECK_HEADER([gcrypt.h], [
         AC_DEFINE([HAVE_CURL_GNUTLS], [1], [Have cURL with GnuTLS support])
       ])
-    ], [
-      AC_MSG_RESULT([no])
-    ], [
-      AC_MSG_RESULT([no])
     ])
 
     CFLAGS="$save_CFLAGS"
diff -durpN php-7.2.34.orig/ext/iconv/config.m4 php-7.2.34/ext/iconv/config.m4
--- php-7.2.34.orig/ext/iconv/config.m4	2024-05-14 11:43:45.644455222 +0200
+++ php-7.2.34/ext/iconv/config.m4	2024-05-14 11:44:29.801914959 +0200
@@ -50,7 +50,7 @@ if test "$PHP_ICONV" != "no"; then
       AC_MSG_CHECKING([if using GNU libiconv])
       php_iconv_old_ld="$LDFLAGS"
       LDFLAGS="-liconv $LDFLAGS"
-      AC_TRY_RUN([
+      PHP_TRY_RUN([
 #include <$PHP_ICONV_H_PATH>
 int main() {
   printf("%d", _libiconv_version);
@@ -128,7 +128,7 @@ int main() {
     esac
 
     AC_MSG_CHECKING([if iconv supports errno])
-    AC_TRY_RUN([
+    PHP_TRY_RUN([
 #include <$PHP_ICONV_H_PATH>
 #include <errno.h>
 
diff -durpN php-7.2.34.orig/ext/opcache/config.m4 php-7.2.34/ext/opcache/config.m4
--- php-7.2.34.orig/ext/opcache/config.m4	2024-05-14 11:43:45.644455222 +0200
+++ php-7.2.34/ext/opcache/config.m4	2024-05-14 11:44:29.802914969 +0200
@@ -29,7 +29,7 @@ if test "$PHP_OPCACHE" != "no"; then
   AC_CHECK_HEADERS([unistd.h sys/uio.h])
 
   AC_MSG_CHECKING(for sysvipc shared memory support)
-  AC_TRY_RUN([
+  PHP_TRY_RUN([
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <sys/ipc.h>
@@ -101,7 +101,7 @@ int main() {
   AC_MSG_RESULT([$msg])
 
   AC_MSG_CHECKING(for mmap() using MAP_ANON shared memory support)
-  AC_TRY_RUN([
+  PHP_TRY_RUN([
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <sys/mman.h>
@@ -155,7 +155,7 @@ int main() {
   AC_MSG_RESULT([$msg])
 
   AC_MSG_CHECKING(for mmap() using /dev/zero shared memory support)
-  AC_TRY_RUN([
+  PHP_TRY_RUN([
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <sys/mman.h>
@@ -212,7 +212,7 @@ int main() {
   AC_MSG_RESULT([$msg])
 
   AC_MSG_CHECKING(for mmap() using shm_open() shared memory support)
-  AC_TRY_RUN([
+  PHP_TRY_RUN([
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <sys/mman.h>
@@ -283,7 +283,7 @@ int main() {
   AC_MSG_RESULT([$msg])
 
   AC_MSG_CHECKING(for mmap() using regular file shared memory support)
-  AC_TRY_RUN([
+  PHP_TRY_RUN([
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <sys/mman.h>
diff -durpN php-7.2.34.orig/ext/posix/config.m4 php-7.2.34/ext/posix/config.m4
--- php-7.2.34.orig/ext/posix/config.m4	2024-05-14 11:43:45.644455222 +0200
+++ php-7.2.34/ext/posix/config.m4	2024-05-14 11:44:29.802914969 +0200
@@ -29,7 +29,10 @@ int main(int argc, char *argv[])
   ],[
     AC_MSG_RESULT([no, posix_ttyname() will be thread-unsafe])
   ], [
-    AC_MSG_RESULT([no, cannot detect working ttyname_r() when cross compiling. posix_ttyname() will be thread-unsafe])
+    AC_CHECK_FUNCS(ttyname_r,
+      [AC_MSG_RESULT([yes])
+       AC_DEFINE(HAVE_TTYNAME_R, 1, [Whether you have a working ttyname_r])],
+      [AC_MSG_RESULT([no, cannot detect working ttyname_r() when cross compiling. posix_ttyname() will be thread-unsafe])])
   ])
 
   AC_CACHE_CHECK([for utsname.domainname], ac_cv_have_utsname_domainname, [
diff -durpN php-7.2.34.orig/ext/standard/config.m4 php-7.2.34/ext/standard/config.m4
--- php-7.2.34.orig/ext/standard/config.m4	2024-05-14 11:43:45.645455233 +0200
+++ php-7.2.34/ext/standard/config.m4	2024-05-14 12:21:33.822272941 +0200
@@ -4,7 +4,7 @@ dnl
 dnl Check if flush should be called explicitly after buffered io
 dnl
 AC_CACHE_CHECK([whether flush should be called explicitly after a buffered io], ac_cv_flush_io,[
-AC_TRY_RUN( [
+PHP_TRY_RUN( [
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
@@ -62,7 +62,7 @@ if test "$ac_cv_func_crypt" = "no"; then
 fi
 
 AC_CACHE_CHECK(for standard DES crypt, ac_cv_crypt_des,[
-  AC_TRY_RUN([
+  PHP_TRY_RUN([
 #include <stdlib.h>
 #include <string.h>
 
@@ -90,7 +90,7 @@ int main() {
 ])])
 
 AC_CACHE_CHECK(for extended DES crypt, ac_cv_crypt_ext_des,[
-  AC_TRY_RUN([
+  PHP_TRY_RUN([
 #include <stdlib.h>
 #include <string.h>
 
@@ -118,7 +118,7 @@ int main() {
 ])])
 
 AC_CACHE_CHECK(for MD5 crypt, ac_cv_crypt_md5,[
-AC_TRY_RUN([
+PHP_TRY_RUN([
 #include <stdlib.h>
 #include <string.h>
 
@@ -156,7 +156,7 @@ int main() {
 ])])
 
 AC_CACHE_CHECK(for Blowfish crypt, ac_cv_crypt_blowfish,[
-AC_TRY_RUN([
+PHP_TRY_RUN([
 #include <stdlib.h>
 #include <string.h>
 
@@ -191,7 +191,7 @@ int main() {
 ])])
 
 AC_CACHE_CHECK(for SHA512 crypt, ac_cv_crypt_sha512,[
-AC_TRY_RUN([
+PHP_TRY_RUN([
 #include <stdlib.h>
 #include <string.h>
 
@@ -225,7 +225,7 @@ int main() {
 ])])
 
 AC_CACHE_CHECK(for SHA256 crypt, ac_cv_crypt_sha256,[
-AC_TRY_RUN([
+PHP_TRY_RUN([
 #include <stdlib.h>
 #include <string.h>
 
diff -durpN php-7.2.34.orig/sapi/fpm/config.m4 php-7.2.34/sapi/fpm/config.m4
--- php-7.2.34.orig/sapi/fpm/config.m4	2024-05-14 11:43:45.645455233 +0200
+++ php-7.2.34/sapi/fpm/config.m4	2024-05-14 11:44:29.804914990 +0200
@@ -68,7 +68,7 @@ AC_DEFUN([AC_FPM_CLOCK],
   if test "$have_clock_gettime" = "no"; then
     AC_MSG_CHECKING([for clock_get_time])
 
-    AC_TRY_RUN([ #include <mach/mach.h>
+    PHP_TRY_RUN([ #include <mach/mach.h>
       #include <mach/clock.h>
       #include <mach/mach_error.h>
 
@@ -120,7 +120,7 @@ AC_DEFUN([AC_FPM_TRACE],
   if test "$have_ptrace" = "yes"; then
     AC_MSG_CHECKING([whether ptrace works])
 
-    AC_TRY_RUN([
+    PHP_TRY_RUN([
       #include <unistd.h>
       #include <signal.h>
       #include <sys/wait.h>
@@ -268,7 +268,8 @@ AC_DEFUN([AC_FPM_TRACE],
       proc_mem_file=""
       AC_MSG_RESULT([no])
     ], [
-      AC_MSG_RESULT([skipped (cross compiling)])
+      AC_MSG_RESULT([cross compiling for Linux, using 'mem'])
+      proc_mem_file="mem"
     ])
   fi
 
diff -durpN php-7.2.34.orig/Zend/Zend.m4 php-7.2.34/Zend/Zend.m4
--- php-7.2.34.orig/Zend/Zend.m4	2024-05-14 11:43:45.645455233 +0200
+++ php-7.2.34/Zend/Zend.m4	2024-05-14 12:19:59.868412961 +0200
@@ -109,7 +109,7 @@ ZEND_CHECK_FLOAT_PRECISION
 dnl test whether double cast to long preserves least significant bits
 AC_MSG_CHECKING(whether double cast to long preserves least significant bits)
 
-AC_TRY_RUN([
+PHP_TRY_RUN([
 #include <stdlib.h>
 #include <limits.h>
 
@@ -272,7 +272,7 @@ AC_MSG_RESULT(done)
 dnl test for memory allocation using mmap(MAP_ANON)
 AC_MSG_CHECKING(for memory allocation using mmap(MAP_ANON))
 
-AC_TRY_RUN([
+PHP_TRY_RUN([
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
@@ -317,7 +317,7 @@ int main()
 dnl test for memory allocation using mmap("/dev/zero")
 AC_MSG_CHECKING(for memory allocation using mmap("/dev/zero"))
 
-AC_TRY_RUN([
+PHP_TRY_RUN([
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
