--- php-5.6.3/acinclude.m4~	2014-11-26 20:22:16.000000000 +0100
+++ php-5.6.3/acinclude.m4	2014-11-26 20:48:47.472043550 +0100
@@ -1905,13 +1905,6 @@
   ])
 
   if test "$found" = "yes"; then
-    ac_libs=$LIBS
-    LIBS="$LIBS -l$2"
-    AC_TRY_RUN([main() { return (0); }],[found=yes],[found=no],[found=no])
-    LIBS=$ac_libs
-  fi
-
-  if test "$found" = "yes"; then
     PHP_ADD_LIBRARY($2)
     PHP_DEF_HAVE($1)
     PHP_DEF_HAVE(lib$2)
