diff -durpN php-7.2.16.orig/acinclude.m4 php-7.2.16/acinclude.m4
--- php-7.2.16.orig/acinclude.m4	2019-03-26 09:58:54.558584186 -0400
+++ php-7.2.16/acinclude.m4	2019-03-26 10:01:04.275226343 -0400
@@ -2268,7 +2268,6 @@ AC_DEFUN([PHP_SETUP_ICU],[
       fi
 
       ICU_VERSION=$icu_version
-      ICU_INCS=`$ICU_CONFIG --cppflags-searchpath`
       ICU_LIBS=`$ICU_CONFIG --ldflags --ldflags-icuio`
       PHP_EVAL_INCLINE($ICU_INCS)
       PHP_EVAL_LIBLINE($ICU_LIBS, $1)
