diff -durpN php-7.3.3.orig/acinclude.m4 php-7.3.3/acinclude.m4
--- php-7.3.3.orig/acinclude.m4	2019-03-27 05:45:53.177711543 -0400
+++ php-7.3.3/acinclude.m4	2019-03-27 05:49:53.440687873 -0400
@@ -2460,7 +2460,8 @@ AC_DEFUN([PHP_SETUP_ICONV], [
   dnl Check libc first if no path is provided in --with-iconv
   dnl
 
-  if test "$PHP_ICONV" = "yes"; then
+  dnl must check against no, not against yes as PHP_ICONV can also include a path, which implies yes
+  if test "$PHP_ICONV" != "no"; then
     dnl Reset LIBS temporarily as it may have already been included
     dnl -liconv in.
     LIBS_save="$LIBS"
diff -durpN php-7.3.3.orig/ext/iconv/config.m4 php-7.3.3/ext/iconv/config.m4
--- php-7.3.3.orig/ext/iconv/config.m4	2019-03-05 08:50:42.000000000 -0500
+++ php-7.3.3/ext/iconv/config.m4	2019-03-27 05:49:53.440687873 -0400
@@ -13,7 +13,7 @@ if test "$PHP_ICONV" != "no"; then
 
   if test "$iconv_avail" != "no"; then
     if test -z "$ICONV_DIR"; then
-      for i in /usr/local /usr; do
+      for i in $PHP_ICONV /usr/local /usr; do
         if test -f "$i/include/iconv.h" || test -f "$i/include/giconv.h"; then
           PHP_ICONV_PREFIX="$i"
           break
