diff -durpN php-7.3.0/acinclude.m4 php-7.3.0.new/acinclude.m4
--- php-7.3.0/acinclude.m4	2019-01-04 09:38:27.054512662 -0500
+++ php-7.3.0.new/acinclude.m4	2019-01-04 09:59:49.196831299 -0500
@@ -1929,13 +1929,6 @@ AC_DEFUN([PHP_CHECK_FUNC_LIB],[
   ])
 
   if test "$found" = "yes"; then
-    ac_libs=$LIBS
-    LIBS="$LIBS -l$2"
-    AC_RUN_IFELSE([AC_LANG_SOURCE([[main() { return (0); }]])],[found=yes],[found=no],[found=no])
-    LIBS=$ac_libs
-  fi
-
-  if test "$found" = "yes"; then
     PHP_ADD_LIBRARY($2)
     PHP_DEF_HAVE($1)
     PHP_DEF_HAVE(lib$2)
