diff -durpN php-7.3.0.1/ext/xsl/config.m4 php-7.3.0.2/ext/xsl/config.m4
--- php-7.3.0.1/ext/xsl/config.m4	2018-12-04 11:12:27.000000000 -0500
+++ php-7.3.0.2/ext/xsl/config.m4	2019-01-04 05:49:17.462106798 -0500
@@ -14,12 +14,7 @@ if test "$PHP_XSL" != "no"; then
     AC_MSG_ERROR([XSL extension requires DOM extension, add --enable-dom])
   fi
 
-  for i in $PHP_XSL /usr/local /usr; do
-    if test -x "$i/bin/xslt-config"; then
-      XSLT_CONFIG=$i/bin/xslt-config
-      break
-    fi
-  done
+  AC_PATH_PROG(XSLT_CONFIG, xslt-config)
 
   if test -z "$XSLT_CONFIG"; then
     AC_MSG_ERROR([xslt-config not found. Please reinstall the libxslt >= 1.1.0 distribution])
