diff -durpN php-7.3.3.orig/ext/ldap/php_ldap.h php-7.3.3/ext/ldap/php_ldap.h
--- php-7.3.3.orig/ext/ldap/php_ldap.h	2019-03-27 06:36:57.072465236 -0400
+++ php-7.3.3/ext/ldap/php_ldap.h	2019-03-27 06:34:52.513517325 -0400
@@ -25,7 +25,7 @@
 #include <lber.h>
 #endif
 
-#include <ldap.h>
+#include "/usr/include/ldap.h"
 
 extern zend_module_entry ldap_module_entry;
 #define ldap_module_ptr &ldap_module_entry
diff -durpN php-7.3.3.orig/ext/oci8/config.m4 php-7.3.3/ext/oci8/config.m4
--- php-7.3.3.orig/ext/oci8/config.m4	2019-03-27 06:36:57.072465236 -0400
+++ php-7.3.3/ext/oci8/config.m4	2019-03-27 06:34:52.514517333 -0400
@@ -372,6 +372,7 @@ if test "$PHP_OCI8" != "no"; then
 
     dnl Header directory for Instant Client SDK RPM install
     OCISDKRPMINC=`echo "$PHP_OCI8_INSTANT_CLIENT" | $PHP_OCI8_SED -e 's!^/usr/lib/oracle/\(.*\)/client\('${PHP_OCI8_IC_LIBDIR_SUFFIX}'\)*/lib[/]*$!/usr/include/oracle/\1/client\2!'`
+    OCISDKRPMINC=`echo "$PHP_OCI8_INSTANT_CLIENT" | $PHP_OCI8_SED -e 's!^/usr/\(lib64\|lib\)/oracle/\(.*\)/\(client64\|client\)/lib[/]*$!/usr/include/oracle/\2/\3!'`
 
     dnl Header directory for Instant Client SDK zip file install
     OCISDKZIPINC=$PHP_OCI8_INSTANT_CLIENT/sdk/include
diff -durpN php-7.3.3.orig/ext/pdo_oci/config.m4 php-7.3.3/ext/pdo_oci/config.m4
--- php-7.3.3.orig/ext/pdo_oci/config.m4	2019-03-27 06:36:57.072465236 -0400
+++ php-7.3.3/ext/pdo_oci/config.m4	2019-03-27 06:34:52.514517333 -0400
@@ -92,7 +92,7 @@ if test "$PHP_PDO_OCI" != "no"; then
 
     AC_MSG_CHECKING([for oci.h])
     dnl Header directory for Instant Client SDK RPM install
-    OCISDKRPMINC=`echo "$PDO_OCI_LIB_DIR" | $PHP_PDO_OCI_SED -e 's!^\(.*\)/lib/oracle/\(.*\)/\('${PDO_OCI_CLIENT_DIR}'\)/lib[/]*$!\1/include/oracle/\2/\3!'`
+    OCISDKRPMINC=`echo "$PDO_OCI_LIB_DIR" | $PHP_PDO_OCI_SED -e 's!^\(.*\)/\(lib64\|lib\)/oracle/\(.*\)/\('${PDO_OCI_CLIENT_DIR}'\)/lib[/]*$!\1/include/oracle/\3/\4!'`
 
     dnl Header directory for manual installation
     OCISDKMANINC=`echo "$PDO_OCI_LIB_DIR" | $PHP_PDO_OCI_SED -e 's!^\(.*\)/lib[/]*$!\1/include!'`
