Upstream-Status: Inappropriate [cross compile specific]
--- php-7.4.11/build/php.m4.orig	2020-10-09 07:09:50.588538423 -0400
+++ php-7.4.11/build/php.m4	2020-10-09 07:10:52.022988805 -0400
@@ -1950,7 +1950,7 @@
   echo > ext/iconv/php_iconv_supports_errno.h
 
   dnl Check libc first if no path is provided in --with-iconv.
-  if test "$PHP_ICONV" = "yes"; then
+  if test "$PHP_ICONV" != "no"; then
     dnl Reset LIBS temporarily as it may have already been included -liconv in.
     LIBS_save="$LIBS"
     LIBS=
--- php-7.4.11/ext/iconv/config.m4.orig	2020-10-09 07:10:07.678663704 -0400
+++ php-7.4.11/ext/iconv/config.m4	2020-10-09 07:11:31.985281816 -0400
@@ -14,7 +14,7 @@
 
   if test "$iconv_avail" != "no"; then
     if test -z "$ICONV_DIR"; then
-      for i in /usr/local /usr; do
+      for i in $PHP_ICONV /usr/local /usr; do
         if test -f "$i/include/iconv.h" || test -f "$i/include/giconv.h"; then
           PHP_ICONV_PREFIX="$i"
           break
